.github/workflows/reusable_on_release.yaml

```diff
@@ -0,0 +1,122 @@
+name: Release pipeline
+
+# Example usage from the caller workflow:
+#
+# It requires the following secrets (inheriting them is recommended for simplicity)
+#  - OHAI_AWS_ROLE_ARN_PRODUCTION
+#  - OHAI_AWS_ACCESS_KEY_ID_PRODUCTION
+#  - OHAI_AWS_SECRET_ACCESS_KEY_PRODUCTION
+#  - OHAI_GPG_PASSPHRASE
+#  - OHAI_GPG_PRIVATE_KEY_BASE64
+#  - COREINT_SLACK_TOKEN
+#  - COREINT_SLACK_CHANNEL
+
+# jobs:
+#   job_name:
+#     uses: newrelic/coreint-automation/.github/workflows/trigger_prerelease.yaml@v1
+#     with:
+#       tag: ${{ github.event.release.tag_name }}
+#       integration_name: "integration-name" # without the 'nri-' prefix
+#       # test_package: false # this will skip the package testing, useful when the integration is included with the agent (nri-docker)
+#       # setup a schema for the publish action different from "ohi"
+#       # publish_schema: "custom"
+#       # setup a custom schema url (requires publish_schema='custom')
+#       # publish_schema_url: "https://raw.githubusercontent.com/newrelic/nri-${{ env.INTEGRATION }}/${{ env.TAG }}/build/s3-publish-schema.yml"
+#     secrets: inherit
+
+on:
+  workflow_call:
+
+    inputs:
+      tag:
+        description: Tag to be released
+        type: string
+        required: true
+      integration_name:
+        description: Integration name to be released (without the 'nri-' prefix)
+        type: string
+        required: true
+      # Optional flags
+      test_package:
+        description: Defines if the package should be tested using the 'newrelic/integrations-pkg-test-action/linux' action.
+        type: boolean
+        required: false
+        default: true
+      publish_schema:
+        description: Defines the schema for the publish_action.
+        type: string
+        required: false
+        default: "ohi"
+      publish_schema_url:
+        description: Defines the schema_url for the publish_action (it requires publish_schema='custom')
+        type: string
+        required: false
+        default: ""
+
+    secrets: # uppercase secrets to match the already defined organization secrets
+      OHAI_AWS_ROLE_ARN_PRODUCTION:
+        required: true
+      OHAI_AWS_ACCESS_KEY_ID_PRODUCTION:
+        required: true
+      OHAI_AWS_SECRET_ACCESS_KEY_PRODUCTION:
+        required: true
+      OHAI_GPG_PASSPHRASE:
+        required: true
+      OHAI_GPG_PRIVATE_KEY_BASE64:
+        required: true
+      COREINT_SLACK_TOKEN:
+        required: true
+      COREINT_SLACK_CHANNEL:
+        required: true
+
+jobs:
+  publish-to-s3:
+    name: Send release assets to S3
+    runs-on: ubuntu-latest
+    steps:
+      - name: Publish to S3 action
+        uses: newrelic/infrastructure-publish-action@v1
+        env:
+          AWS_S3_BUCKET_NAME: "nr-downloads-main"
+          AWS_S3_LOCK_BUCKET_NAME: "onhost-ci-lock"
+          AWS_REGION: "us-east-1"
```

Comment:
are these needed as env?

Reply:
I do not think so, I am checking the code, but the [action ](https://github.com/newrelic/infrastructure-publish-action/blob/main/action.yml) is calling directly with all the inputs a shell script that calls [docker](https://github.com/newrelic/infrastructure-publish-action/blob/1ef5c3e4e9cc90c68b88cfec54d521d769877da8/action-run.sh#L34) from the envs that were passed from the action itself. 
Therefore, I am removing them